File: leet-code/minTimeToVisitAllPoints.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function (points) {
  let total = 0
  for (let i = 1; i < points.length; i++) {
    let x = 0
    let y = 0
    x = Math.abs(points[i][0] - points[i - 1][0])
    y = Math.abs(points[i][1] - points[i - 1][1])
    total += x > y ? x : y
  }
  return total
}

File: leet-code/minTimeToVisitAllPoints.go
package main

func minTimeToVisitAllPoints(points [][]int) int {
	var total int
	for i := 1; i < len(points); i++ {
		val := points[i]
		prev := points[i-1]
		x, y := 0, 0
		x = val[0] - prev[0]
		y = val[1] - prev[1]
		if x < 0 {
			x = -x
		}
		if y < 0 {
			y = -y
		}
		if x > y {
			total += x
		} else {
			total += y
		}
	}
	return total
}
